From f2cef795251716b931465ba34bc221d759db6120 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 28 Oct 2022 13:53:08 +0200
Subject: [PATCH] Remove Werror from cflags

---
 pyobjc_setup.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/pyobjc_setup.py b/pyobjc_setup.py
index f6415b3..d073bcc 100644
--- a/pyobjc_setup.py
+++ b/pyobjc_setup.py
@@ -506,9 +506,6 @@ def Extension(*args, **kwds):
     if os_level == "10.4":
         cflags.append("-DNO_OBJC2_RUNTIME")
 
-    if "-Werror" not in cflags:
-        cflags.append("-Werror")
-
     if "extra_compile_args" in kwds:
         kwds["extra_compile_args"] = kwds["extra_compile_args"] + cflags
     else:
-- 
2.37.0 (Apple Git-136)

