diff --git a/pyobjc-framework-Cocoa/pyobjc_setup.py b/pyobjc-framework-Cocoa/pyobjc_setup.py
index 39d9889cb..2b685f3f7 100644
--- a/pyobjc-framework-Cocoa/pyobjc_setup.py
+++ b/pyobjc-framework-Cocoa/pyobjc_setup.py
@@ -202,6 +202,32 @@ def get_os_level():
     return ".".join(v.split(".")[:2])
 
 
+if "-Wl" in get_config_var("CFLAGS"):
+    # some Python versions have "-Wl,-export_dynamic" present, causing
+    # Clang to complain
+    print("PATCH: remove linker flags...")
+    config_vars = get_config_vars()
+    for k in ('CFLAGS', 'CPPFLAGS'):
+        if isinstance(config_vars[k], str) and "-Wl" in config_vars[k]:
+            flags = config_vars[k].split()
+            new_flags = []
+            for f in flags:
+                if f.startswith("-Wl"):
+                    continue
+                else:
+                    new_flags.append(f)
+            config_vars[k] = ' '.join(new_flags)
+            print("  ...fixed:", k, ' '.join(new_flags))
+
+
+# overwrite the linker flags using CC and LDFLAGS set in the build script
+_ldshared = get_config_var("LDSHARED").split()
+_ldshared[0] = os.environ['CC']
+_ldshared = ' '.join(_ldshared) + ' ' + os.environ['LDFLAGS'] + ' ' + os.environ['CPPFLAGS']
+_ldshared = _ldshared.replace("-Wl,-dead_strip_dylibs", "")
+get_config_vars()["LDSHARED"] = _ldshared
+
+
 def get_sdk():
     env_cflags = os.environ.get("CFLAGS", "")
     config_cflags = get_config_var("CFLAGS")
